lite2/verifier.go

```diff
@@ -0,0 +1,116 @@
+package lite
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/pkg/errors"
+
+	cmn "github.com/tendermint/tendermint/libs/common"
+	"github.com/tendermint/tendermint/types"
+)
+
+var (
+	// DefaultTrustLevel - new header can be trusted if at least one correct old
+	// validator signed it.
+	DefaultTrustLevel = cmn.Fraction{Numerator: 1, Denominator: 3}
+)
+
+func Verify(
+	chainID string,
+	h1 *types.SignedHeader,
+	h1NextVals *types.ValidatorSet,
+	h2 *types.SignedHeader,
+	h2Vals *types.ValidatorSet,
+	trustingPeriod time.Duration,
+	now time.Time,
+	trustLevel cmn.Fraction) error {
+
+	if err := ValidateTrustLevel(trustLevel); err != nil {
+		return err
+	}
+
+	// Ensure last header can still be trusted.
```

Comment:
This should probably be extracted into a function.

Reply:
why?

Reply:
Defining what is a trusted header will potentially used from several places. It is also important part of business logic that could be changed in the future so it is probably good idea if it's isolated in a function.

Reply:
"will potentially used" argument rarely works in practice. better to make changes when A/B/C actually happens